test/evp_test.c

```diff
@@ -660,6 +660,36 @@ static int cipher_test_enc(EVP_TEST *t, int enc,
         /* We supply the data all in one go */
         if (!EVP_CipherUpdate(ctx, tmp + out_misalign, &tmplen, in, in_len))
             goto err;
+    } else if (frag == 2) {
+        if (EVP_CIPHER_CTX_block_size(ctx) > 1
+                && in_len >= (size_t)EVP_CIPHER_CTX_block_size(ctx)) {
+            if (!EVP_CipherUpdate(ctx, tmp + out_misalign, &chunklen, in,
+                                  EVP_CIPHER_CTX_block_size(ctx) - 1))
+                goto err;
+            tmplen += chunklen;
+            in += EVP_CIPHER_CTX_block_size(ctx) - 1;
+            in_len -= EVP_CIPHER_CTX_block_size(ctx) - 1;
+        }
+        if (in_len > 1) {
+            /* this is expected to fail */
+            memcpy(tmp + out_misalign + tmplen, in, 2);
+            if (!EVP_CipherUpdate(ctx, tmp + out_misalign + tmplen, &chunklen,
+                                  tmp + out_misalign + tmplen, 2)) {
+               ERR_clear_error();
```

Comment:
Wouldn't it be more appropriate to check for specific error? As opposite to ignoring *any* that is.

Reply:
And in case test fails [i.e. EVP_CipherUpdate returning success], wouldn't it be appropriate to set t->err to something specific and terminate test? I mean as it stands now, failure at this point would hopefully map to VALUE_MISMATCH, but isn't not actually given that it does for *every* given test.

Reply:
Yeah, this is a kind of borderline test here, and it does indeed succeed if the
cipher block size == 1.
In a way the in and out buffers are **fully** overlapping (in == out)
yet we get an EVP_R_PARTIALLY_OVERLAPPING (which is in a way
questionable in this case)
I could change in and out to be off by 1, and then it would always
fail, simply because the buffer is **partially** overlapping.